log4j-slf4j-impl/src/main/java/org/apache/logging/slf4j/Log4jMarkerFactory.java

```diff
@@ -113,26 +125,18 @@ public boolean exists(final String name) {
         return markerMap.containsKey(name);
     }
 
-    /**
-     * Log4j does not support detached Markers. This method always returns false.
-     * @param name The Marker name.
-     * @return {@code false}
-     */
     @Override
     public boolean detachMarker(final String name) {
-        return false;
+        if (name == null) {
+            return false;
+        }
+        detachedMarkers.add(name);
+        return true;
     }
 
-    /**
-     * Log4j does not support detached Markers for performance reasons. The returned Marker is attached.
-     * @param name The Marker name.
-     * @return The named Marker (unmodified).
-     */
     @Override
     public Marker getDetachedMarker(final String name) {
-        LOGGER.warn("Log4j does not support detached Markers. Returned Marker [{}] will be unchanged.", name);
-        return getMarker(name);
+        return new Log4jMarker(new MarkerManager.Log4jMarker(name));
```

Comment:
I don't think this will work correctly when compared against a standard non-detached marker that log4j uses in filters because they use an instance comparison, not a string value comparison.

Reply:
So I think my point is, we may need a different log4j marker implementation than `MarkerManager.Log4jMarker` which handles string-value comparisons rather than identity comparisons.

Reply:
I start to see the problem... let me try to implement such value based marker :)

Reply:
@carterkozak looks like that `MarkerManager.Log4jMarker` is already string-value:

```java 
final Marker marker = (Marker) o;
return name.equals(marker.getName());
```

and Log4J implementation of SLF4J marker is doing:

```
final Log4jMarker other = (Log4jMarker) obj;
		if (!Objects.equals(marker, other.marker)) {
			return false;
		}
		return true;
```

did I miss something?

Reply:
@carterkozak created unit test:

```java
    @Test
    void comparisonDetachedAndNonDetachedMarkers() {
        // Given
        Log4jMarkerFactory sut = new Log4jMarkerFactory();

        // When
        Marker nonDetachedMarker = sut.getMarker("name");
        Marker detachedMarker = sut.getDetachedMarker("name");

        // Then
        assertNotSame(nonDetachedMarker, detachedMarker);
        assertTrue(nonDetachedMarker.equals(detachedMarker));
        assertTrue(detachedMarker.equals(nonDetachedMarker));
    }
```

does that looks good for you? (test is green)

Reply:
Most checks don't use `.equals(`, rather `.isInstanceOf(`, for example https://github.com/apache/logging-log4j2/blob/f72100df0decc9bda96b4d769822c4e48b2848fc/log4j-core/src/main/java/org/apache/logging/log4j/core/filter/MarkerFilter.java#L70-L72

Reply:
@carterkozak but that seems to be fine as well:

```java
        assertTrue(((Log4jMarker) detachedMarker).getLog4jMarker().isInstanceOf("name"));
        assertTrue(((Log4jMarker) nonDetachedMarker).getLog4jMarker().isInstanceOf("name"));
```

Reply:
implementation is simply: 

```java
        @Override
        @PerformanceSensitive({"allocation", "unrolled"})
        public boolean isInstanceOf(final String markerName) {
            requireNonNull(markerName, "A marker name is required");
            if (markerName.equals(this.getName())) {
                return true;
            }
```

Reply:
I think the string method is fine, the overload which takes another marker is not